docs/index.bs

```diff
@@ -3258,6 +3260,38 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
+              1. Run the following substeps [=in parallel=], but [=abort when=] |fetchController|'s [=fetch controller/state=] is "<code>terminated</code>" or "<code>aborted</code>":
+                  1. Set |raceFetchController| to the result of calling [=fetch=] given |request|, with [=fetch/processResponse=] set to the following steps given a [=/response=] |raceNetworkRequestResponse|:
+                      1. If |raceNetworkRequestResponse|'s [=response/status=] is [=ok status=], then:
+                          1. Set |raceResponse|'s [=race response/value=] to |raceNetworkRequestResponse|.
+                          1. [=queue/Enqueue=] |raceNetworkRequestResponse| to |queue|.
+                      1. Otherwise, set |raceResponse|'s [=race response/value=] to a [=network error=].
+              1. [=If aborted=] and |raceFetchController| is not null, then:
+                  1. If |raceFetchController|'s [=fetch controller/state=] is not "<code>terminated</code>", [=fetch controller/abort=] |raceFetchController|.
+                  1. Set |raceResponse| to a [=race response=] whose [=race response/value=] is null.
+              1. Resolve |preloadResponse| with undefined.
+              1. Run the following substeps [=in parallel=]:
+                  1. [=map/For each=] |cacheName| → |cache| of the |registration|'s [=service worker registration/storage key=]'s [=name to cache map=]:
+                      1. If |source|["{{RouterSourceDict/raceNetworkAndCacheCacheName}}"] [=map/exists=] and |source|["{{RouterSourceDict/raceNetworkAndCacheCacheName}}"] [=string/is=] not |cacheName|, [=continue=].
+                      1. Let |requestResponses| be the result of running [=Query Cache=] with |request|, a new {{CacheQueryOptions}}, and |cache|.
+                      1. Let |globalObject| be null.
+                      1. If |requestResponses| is not an empty [=list=], then:
+                          1. Let |requestResponse| be the first element of |requestResponses|.
+                          1. Let |response| be |requestResponse|'s response.
+                          1. If |response|'s [=response/type=] is "`opaque`", and |globalObject| is null:
+                              1. Set |globalObject| to the result of running [=Setup ServiceWorkerGlobalScope=] with |activeWorker|.
+                          1. If |globalObject| is null, [=continue=].
+                          1. If |response|'s [=response/type=] is "`opaque`", and [=cross-origin resource policy check=] with |globalObject|'s [=environment settings object/origin=], |globalObject|, "", and |response|'s [=filtered response/internal response=] returns <b>blocked</b>, then [=continue=].
+                          1. [=queue/Enqueue=] |response| to |queue|.
+                          1. If |raceFetchController| is not null, [=fetch controller/abort=] |raceFetchController|.
+              1. Wait until |queue| is not empty.
+              1. If |queue| is empty, return null.
```

Comment:
What does this step mean?
L3292 says "Wait until |queue| is not empty.", then L3293 should not happen?

Reply:
I guess L3292 might wait for:
1. |queue| is fulfilled
2. or, both of network fetch and cache lookup failure.

But stepping back, race-network-and-fetch-event may also need both failure case.

Reply:
Added flags to track if they are finished and Wait until |queue| is not empty step is updated so it won't wait indefinitely if both fails. The next step handles if they fail and |queue| is empty our algorithm returns.

Is this change, okay? Feel free to suggest for corrections or if it is unclear

Reply:
Thanks for the update.
The algorithm looks good except for the abort network scenario.
I want a spec expert to check the way to write the procedure down to specification.